### PR TITLE
docs(troubleshooting): Side effects with subscribe

### LINKED ISSUE #331: Troubleshooting docs: epics fail silently when returning subscribe from epic
When you return a subscribe from an epic, it just fails silently and none of the epics execute. I'm going to submit a PR with an example on how to do side effects without using `subscribe`.

### PR DESCRIPTION
**Pull request by hendrikswan**
closes #331

- [x] I have read the [Contributor Guide](https://github.com/redux-observable/redux-observable/blob/master/CONTRIBUTING.md)
- [x] My commit messages are in [conventional-changelog-standard](https://github.com/redux-observable/redux-observable/blob/master/CONTRIBUTING.md#sending-a-pull-request) format.

### CODE REVIEW

File: docs/Troubleshooting.md [hunk 1]
```diff
@@ -95,6 +95,28 @@ class TooFancy {
...
+You might be doing a `subscribe` in one of your epics: 
+
+```js
+const myEpic = action$ => { // performing side effect with .subscribe!
+  return action$.ofType(...)
+  .subscribe(item => console.log(item));
+};
+```
+
+This happens because subscribe doesn't return a promise itself. When you want to perform side effects based on actions, without the need to return an observable from your epic containing downstream actions, you can follow this pattern:
```
Comment (jayphelps) [inline docs/Troubleshooting.md]: > This happens because subscribe doesn't return a promise itself.

Can you clarify what you mean here?

> without the need to return an observable from your epic containing downstream actions

Epics must always return an Observable, and indeed you are--it's just an Observable that never emits anything. An important distinction.

**Comment by jayphelps** [comment]
Cross posting here: https://github.com/redux-observable/redux-observable/issues/331#issuecomment-332957470

> This is unfortunately another example of #263 and https://github.com/ReactiveX/rxjs/issues/2813. There is an error about not providing an Observable where it was expected, it's just being swallowed silently due to a bug.
> 
> As https://github.com/ReactiveX/rxjs/issues/2813 mentions though, while this is a bug, in the future this behavior will likely be intentional, though almost certainly with a better debugging experience by still showing the error in the debug console (which is what most people actually care about when they freak out about this change lol).

**Comment by hendrikswan** [comment]
Yeah, good idea, I'll link to the reason why it happens.

**Comment by hendrikswan** [comment]
Hey @jayphelps, I linked to the issue as you suggested. Everything you linked to seems to expound on swallowing of errors. 

Maybe I'm missing something, and I wonder if it also wouldn't overwhelm other `Observable` laymen like me, but why didn't my scenario just result in this error? `TypeError: object is not observable`. 

It's as if the result of calling `subscribe` looks enough like an `Observable` and fools the logic that checks whether it is one, and we then proceed to work with it as if it's an `Observable`, which in turn causes errors to be swallowed. 

Isn't there something that we can do in redux-observable, so that we can catch this scenario and provide a specific error? Something like: "You did not return an observable from your epic".

**Comment by hendrikswan** [comment]
And if there is something that we can do, I'd be happy to do a PR for that.

**Comment by jayphelps** [comment]
Thanks @hendrikswan ! 🎉

**Comment by hendrikswan** [comment]
Thanks for your patience @jayphelps :)

**Comment by jayphelps** [comment]
> Maybe I'm missing something, and I wonder if all this information wouldn't also overwhelm other Observable laymen like me, but why didn't my scenario just result in this error? TypeError: object is not observable.

@hendrikswan oh shoot I wrote a nice reply to that question... I must have forgot to send it.. Sorry about that.

tl;dr is that there is in fact an error, it's just being [silently swallowed by rxjs](https://github.com/ReactiveX/rxjs/issues/2813). You can see the error by placing a `.do({ error: e => console.error(e) })` in your root epic

```js
const rootEpic = (...args) =>
  combineEpics(...yourEpicsHere)(...args)
    .do({ error: e => console.error(e) });
```

The error is just the normal:

> You provided and invalid object where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.

**Comment by hendrikswan** [comment]
@jayphelps, ah now I understand! Thanks a lot for taking the time to explain it 👍
